src/lib/Util.ts

```diff
@@ -209,6 +209,44 @@ export class Util {
             });   
         });            
     }
+    public static async getMasterProductsNames(productFileNames: Set<String>) {
+        let techIds = new Set<String>(); 
+
+        await Promise.all([...productFileNames].map(async productFileName => {
+            const product = await this.readProduct(productFileName);
+            const masterProductTechIds = product['attributeValueDependencies'].reduce((acc, dependency)=>{
+                const masterProduct = dependency['enxCPQ__Master_Product__r'];
+                if(masterProduct !=null && masterProduct['enxCPQ__TECH_External_Id__c'] !=null){
+                    return [...acc, masterProduct['enxCPQ__TECH_External_Id__c']];
+                }
+                return acc;
+            }, []);
+            techIds = new Set<String>([...techIds, ...masterProductTechIds]);
+            return;
+        }));
+
+        const allProducts = await this.readAllFiles('/products');
+        return new Set(allProducts.reduce((acc, product) => {
+            const root = product['root'];
+            if(techIds.has(root['enxCPQ__TECH_External_Id__c']) && root['Name']){
+                return [...acc, root['Name']];
+            }
+            return acc;
+        }, []));
+        
+    }
+
+    public static async getMasterProductsFileNames(productNames: Set<string>) {
+        
+        let masterProductsFileNames = new Set<String>();
+        for (let productName of productNames){
```

Comment:
Could be reduce or map. Also new Set can be done on the result of map/reduce and not in loop